src/reporters/IReporter.ts

```diff
@@ -17,7 +18,7 @@ export interface IReporter {
   buildReport(
     practicesAndComponents: PracticeWithContextForReporter[],
     practicesAndComponentsAfterFix?: PracticeWithContextForReporter[],
-  ): string | JSONReport;
+  ): string | JSONReport | ReportData[];
```

Comment:
It should be a JSON or Object?

Reply:
I don't know. It could be better to return an Object because we could add some global info to the root of report. E.g. scan duration etc.